_posts/2016-10-05-writing-fast-rails.md

```diff
@@ -0,0 +1,133 @@
+
+    Order.where(number: 'R545612547').map &:id
+    # Order Load (5.0ms)  SELECT `orders`.* FROM `orders` WHERE `orders`.`number` = 'R545612547' ORDER BY orders.created_at DESC
+    => [1]
+
+As with `select`, `map` will load the `order` into memory and it will get the
+`id` attribute.
+
+Using `pluck` will be faster, because it doesn't need to load an entire object
+into memory.
+
+So this will be much faster:
+
+    Order.where(number: 'R545612547').pluck :id
+    # SQL (0.8ms)  SELECT `orders`.`id` FROM `orders` WHERE `orders`.`number` = 'R545612547' ORDER BY orders.created_at DESC
+    => [1]
+
+For this particular case, `pluck` six times faster than `map`.
+
+## Avoid N+1 Queries
+
+There are some rare cases where you want an **N+1 query** in your application.
+For  instance, when you are using a
+[Russian Doll Caching](http://edgeguides.rubyonrails.org/caching_with_rails.html#russian-doll-caching)
+strategy, it's a good idea. (full explanation in this interview with [DHH](https://twitter.com/dhh):
+  [https://youtu.be/ktZLpjCanvg?t=4m27s](https://youtu.be/ktZLpjCanvg?t=4m27s))
+
+If you are **not** using this caching strategy, you should get rid of all your
+[N+1 query problems](http://guides.rubyonrails.org/active_record_querying.html#eager-loading-associations)
+by including the tables that you need before running the query.
+
+For example:
+
+    Order.where("created_at > ?", 1.hour.ago)
+         .find_each do |order|
+      puts order.ship_address.try(:firstname)
+    end
+      Order Load (7866.0ms)  SELECT `orders`.* FROM `orders` WHERE (created_at > '2016-10-05 18:05:48') ORDER BY `orders`.`id` ASC LIMIT 1000
+      Address::ShipAddress Load (0.5ms)  SELECT `addresses`.* FROM `addresses` WHERE `addresses`.`type` IN ('Address::ShipAddress') AND `addresses`.`order_id` = 2619178 LIMIT 1
+      Address::ShipAddress Load (0.5ms)  SELECT `addresses`.* FROM `addresses` WHERE `addresses`.`type` IN ('Address::ShipAddress') AND `addresses`.`order_id` = 2619179 LIMIT 1
+      Address::ShipAddress Load (0.5ms)  SELECT `addresses`.* FROM `addresses` WHERE `addresses`.`type` IN ('Address::ShipAddress') AND `addresses`.`order_id` = 2619180 LIMIT 1
+      Address::ShipAddress Load (0.5ms)  SELECT `addresses`.* FROM `addresses` WHERE `addresses`.`type` IN ('Address::ShipAddress') AND `addresses`.`order_id` = 2619181 LIMIT 1
+      Address::ShipAddress Load (0.5ms)  SELECT `addresses`.* FROM `addresses` WHERE `addresses`.`type` IN ('Address::ShipAddress') AND `addresses`.`order_id` = 2619182 LIMIT 1
+      # ... to N
+
+This code will perform **one** query on the `orders` table and **N** queries on
+the `addresses` table.
+
+This will be faster:
+
+    Order.eager_load(:ship_address)
+         .where("orders.created_at > ?", 1.hour.ago)
+         .find_each do |order|
+      puts order.ship_address.try(:firstname)
+    end
+
+This code will perform only one query. A good way to find N+1 queries is using [bullet](https://rubygems.org/gems/bullet).
```

Comment:
@etagwerker maybe you can include a reference to `eager_load` here? Either at the beginning, instead of "This will be faster" you could say "Using preloading will be faster", or somewhere around here at the end, "Using preloading your code will perform only one query.". 

You could also throw in a reference to [this post](http://blog.arkency.com/2013/12/rails4-preloading) or a similar one which explains `eager_load` vs `includes`?